tests/nexus/workflow_test.py

```diff
@@ -0,0 +1,314 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+import pytest
+import scipp as sc
+from ess.reduce.nexus import workflow
+from scipp.testing import assert_identical
+
+
+@pytest.fixture(params=[{}, {'aux': 1}])
+def group_with_no_position(request) -> workflow.NeXusSample:
+    return workflow.NeXusSample(sc.DataGroup(request.param))
+
+
+def test_sample_position_returns_position_of_group() -> None:
+    position = sc.vector([1.0, 2.0, 3.0], unit='m')
+    sample_group = workflow.NeXusSample(sc.DataGroup(position=position))
+    assert_identical(workflow.get_sample_position(sample_group), position)
+
+
+def test_get_sample_position_returns_origin_if_position_not_found(
+    group_with_no_position,
+) -> None:
+    assert_identical(
+        workflow.get_sample_position(group_with_no_position), workflow.origin
```

Comment:
Should the sample position be optional or should it raise if not found (like for the source position)? I know it was missing from some of our files, but maybe it's good to raise to indicate there's something wrong in the file?

Reply:
Not sure it is wrong. There can be runs where it is irrelevant, e.g., a polarizer transmission run. I don't feel raising is beneficial? In fact I think we should consider not raising if the is no source position: We already know that in many cases we cannot actually compute TOF from the source position, so we should not require this. I suggest (as listed in the "future work" items in the PR message) we address this in follow-up work.